src/dune_rules/pkg_rules.ml

```diff
@@ -1224,13 +1263,26 @@ let gen_rules context_name (pkg : Pkg.t) =
   in
   rule build_rule
 
+let sys_poll_thunk key ~path =
+  let fiber = key ~path in
+  fun () -> Memo.of_reproducible_fiber fiber
+
 let setup_package_rules context ~dir ~pkg_name :
     Build_config.gen_rules_result Memo.t =
   match Package.Name.of_string_user_error (Loc.none, pkg_name) with
   | Error m -> raise (User_error.E m)
   | Ok name -> (
+    (* TODO: PATH *)
+    let path = [] in
     let* db = Lock_dir.get context in
-    resolve db.packages context name >>| function
+    let sys_poll =
```

Comment:
I think you can probably just make this global and use the path from `Env.initial`.